resolwe/flow/managers/dispatcher.py

```diff
@@ -574,6 +585,7 @@ def override_settings(self, **kwargs):
         elif cmd == WorkerProtocol.FINISH:
             try:
                 data_id = message[WorkerProtocol.DATA_ID]
+                data_location = Data.objects.get(id=data_id).location
```

Comment:
Do the query with `DataLocation` instead of `Location` as it is more efficient (uses less joins).